test/libsolidity/smtCheckerTests/operators/userDefined/user_defined_division_same_as_builtin.sol

```
@@ -2,15 +2,15 @@ type U is uint;
 using {div as /} for U global;
 
 function div(U x, U y) pure returns (U) {
-    return U.wrap(U.unwrap(x) / U.unwrap(y));
+    return U.wrap(U.unwrap(x) / U.unwrap(y)); // detects division by zero
 }
 
 contract C {
     function f(U x, U y) public pure returns (U) {
-        return x / y; // FIXME: should detect div by zero
+        return x / y; // FIXME: are we ok that div by zero not reported here?
```

There is a potential problem that errors are not reported where the operator is used, but in the implementation. 
In this example division by zero is reported in line 5 but not here.
I could live with that, but maybe we want the error to be reported here as well?

---

Oh, it makes perfect sense to report it in the implementation. I guess that's consistent with what would happen for a `div(x, y)` function that does division. So I think this is fine and probably even expected.

---

I agree. The implementation of the user-defined operator can be arbitrary code, so it makes sense to pinpoint the exact problem in the operator implementation.